### PR TITLE
RUN-2766: fix: imported command/script steps should be plugin steps

### PR DESCRIPTION
**Pull request by gschueler**
**Is this a bugfix, or an enhancement? Please describe.**
Fix: imported workflow command, script or script file/url step types are correctly imported as plugin steps.

Problem: importing job definitions with script/command/script file steps, imports the steps as the legacy builtin types, instead of converting to plugin types which were introduced in rundeck 5.0.  The workflow editor GUI then falls back to the legacy UI for modifying these step types.

**Describe the solution you've implemented**

Update the workflow import to correctly import the steps as plugin steps.

**Describe alternatives you've considered**

**Additional context**

https://github.com/rundeck/rundeck/pull/8520

### CODE REVIEW

**Review by smartinellibenedetti — APPROVED** [review]
I confirmed the fix by doing the steps described on slack (create a job with a command and a script step, export and import again), both on staging and locally. 

And I also tested the changes by importing one of the xmls defined in this PR in both environments.

I left a few questions/suggestions but no blockers.

File: rundeckapp/grails-app/domain/rundeck/CommandExec.groovy [hunk 1]
```diff
@@ -193,45 +196,90 @@ public class CommandExec extends WorkflowStep implements BaseCommandExec {
...
+     */
+    static String getLegacyBuiltinCommandType(Map data){
+        if (data.exec != null) {
+            return ExecCommand.EXEC_COMMAND_TYPE
+        } else if (data.script != null) {
+            return ScriptCommand.SCRIPT_COMMAND_TYPE
+        } else if (data.scriptfile != null || data.scripturl!=null) {
+            return ScriptFileCommand.SCRIPT_FILE_COMMAND_TYPE
+        }
+    }
```
[the remark this answers is not in the document]
Comment (gschueler) [reply rundeckapp/grails-app/domain/rundeck/CommandExec.groovy:248]: I will add an exception throw in that case, it should never happen due to the isLegacyBuiltinCommandData method check, so shouldn't hurt to add that.

File: rundeckapp/src/test/groovy/rundeck/JobsXMLCodecSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,870 @@
...
+import rundeck.codecs.JobsXMLCodec
+import spock.lang.Specification
+
+
+class JobsXMLCodecSpec extends Specification {
+
+    def "testDecodeBasic2"() {
+        given:
+
+            def xml = """<joblist>
```
[the remark this answers is not in the document]
Comment (gschueler) [reply rundeckapp/src/test/groovy/rundeck/JobsXMLCodecSpec.groovy:16]: I guess it's a trade off: with the definition in the code, I think it's easier to refer to in relation to the test code.

File: rundeckapp/src/test/groovy/rundeck/JobsYAMLCodecSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,600 @@
...
+            cmd3.jobGroup == "job group"
+            cmd3.argString == "job args"
+
+            cmd3.errorHandler
+            cmd3.errorHandler.class == JobExec.class
+            cmd3.errorHandler.jobName == "err job name"
+            cmd3.errorHandler.jobGroup == "err job group"
+            cmd3.errorHandler.argString == "err job args"
+            cmd3.errorHandler.keepgoingOnSuccess
+    }
```
[the remark this answers is not in the document]
Comment (gschueler) [reply rundeckapp/src/test/groovy/rundeck/JobsYAMLCodecSpec.groovy:599]: will add some additional tests

**Pull request merged**